pkg/stub/handler.go

```diff
@@ -460,31 +561,36 @@ func (h *Handler) processFiles(dir string, files []os.FileInfo, opcfg *v1alpha1.
 
 func (h *Handler) updateDockerPullSpec(oldies []string, imagestream *imagev1.ImageStream, opcfg *v1alpha1.SamplesResource) {
 	if len(opcfg.Spec.SamplesRegistry) > 0 {
-		replaced := false
-		for _, old := range oldies {
-			if strings.HasPrefix(imagestream.Spec.DockerImageRepository, old) {
-				imagestream.Spec.DockerImageRepository = strings.Replace(imagestream.Spec.DockerImageRepository, old, opcfg.Spec.SamplesRegistry, 1)
-				replaced = true
-				break
+		if !strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
+			// if not one of our 4 defaults ...
+			replaced := false
+			for _, old := range oldies {
+				if strings.HasPrefix(imagestream.Spec.DockerImageRepository, old) {
+					imagestream.Spec.DockerImageRepository = strings.Replace(imagestream.Spec.DockerImageRepository, old, opcfg.Spec.SamplesRegistry, 1)
+					replaced = true
+					break
+				}
+			}
+			if !replaced {
+				imagestream.Spec.DockerImageRepository = opcfg.Spec.SamplesRegistry + "/" + imagestream.Spec.DockerImageRepository
```

Comment:
i'm confused by this.  

if my DockerImageRepository value is "myprivateregistry/myrepo/myimage"

you're going to replace it with "samplesregistry/myprivateregistry/myrepo/myimage"?

what's the intent here?

Same question about the logic for tags below.

Reply:
I was trying to distinguish between myprivateregistry/myrepo/myimage and myreop/myimage

Though at one point I had code to count the "/" characters as a way to distinguish between the two and that got lost in the shuffle

Now, that said, we are calling this function on the imagestream read from the file

If we are always confident that content from openshift/library will have myregistry/myrepo/myimage, then I'll forgo fixing my attempt to address the 2 and 1 "/" cases, and just simplify

thoughts?